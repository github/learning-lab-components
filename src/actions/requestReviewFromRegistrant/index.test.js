const requestReviewFromRegistrant = require('./index')
const mockContext = require('../../tests/mockContext')

describe('requestReviewFromRegistrant', () => {
  let context

  beforeEach(() => {
    context = mockContext({
      pull_request: {
        number: 1
      }
    }, {
      pullRequests: {
        createReviewRequest: jest.fn()
      },
      search: {
        issues: jest.fn(() => Promise.resolve({ data: { items: [{ title: 'A pr title', number: 2 }] } }))
      }
    })
  })

  it('requests a review on the PR in the payload', async () => {
    await requestReviewFromRegistrant(context, {})
    expect(context.github.pullRequests.createReviewRequest).toHaveBeenCalled()
    expect(context.github.pullRequests.createReviewRequest.mock.calls).toMatchSnapshot()
  })

  it('requests a review on the PR with the specified number', async () => {
    await requestReviewFromRegistrant(context, { pullRequest: 2 })
    expect(context.github.pullRequests.createReviewRequest).toHaveBeenCalled()
    expect(context.github.pullRequests.createReviewRequest.mock.calls).toMatchSnapshot()
  })

  it('requests a review on the PR with the specified title', async () => {
    await requestReviewFromRegistrant(context, { pullRequest: 'A pr title' })
    expect(context.github.pullRequests.createReviewRequest).toHaveBeenCalled()
    expect(context.github.pullRequests.createReviewRequest.mock.calls).toMatchSnapshot()
  })
})
