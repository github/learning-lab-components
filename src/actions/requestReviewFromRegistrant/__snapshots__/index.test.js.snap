// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requestReviewFromRegistrant requests a review on the PR in the payload 1`] = `
Array [
  Array [
    Object {
      "number": 1,
      "owner": "JasonEtco",
      "repo": "example",
      "reviewers": Array [
        "JasonEtco",
      ],
    },
  ],
]
`;

exports[`requestReviewFromRegistrant requests a review on the PR with the specified number 1`] = `
Array [
  Array [
    Object {
      "number": 2,
      "owner": "JasonEtco",
      "repo": "example",
      "reviewers": Array [
        "JasonEtco",
      ],
    },
  ],
]
`;

exports[`requestReviewFromRegistrant requests a review on the PR with the specified title 1`] = `
Array [
  Array [
    Object {
      "number": 2,
      "owner": "JasonEtco",
      "repo": "example",
      "reviewers": Array [
        "JasonEtco",
      ],
    },
  ],
]
`;
